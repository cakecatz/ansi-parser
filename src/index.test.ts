import { parseAnsi } from "./index";
import * as chalk from "chalk";

const ctx = new chalk.Instance({ level: 1 });

test("parse empty string", () => {
  expect(parseAnsi("")).toEqual([]);
});

test("parse string not included ansi codes", () => {
  expect(parseAnsi("hello,world")).toMatchSnapshot();
});

test("string with ansi color", () => {
  expect(parseAnsi(ctx.blue("hello, world"))).toMatchSnapshot();
});

test("bold modifier", () => {
  expect(parseAnsi(ctx.bold("hello, world"))).toMatchSnapshot();
});

test("underline modifier", () => {
  expect(parseAnsi(ctx.underline("hello, world"))).toMatchSnapshot();
});

test("italic modifier", () => {
  expect(parseAnsi(ctx.italic("hello, world"))).toMatchSnapshot();
});

test("inverse modifier", () => {
  expect(parseAnsi(ctx.inverse("hello, world"))).toMatchSnapshot();
});

test("multiple modifiers and colors", () => {
  expect(
    parseAnsi(
      "START" +
        ctx.inverse("INVERSE") +
        ctx.blue("BLUE") +
        ctx.underline("UNDERLINE") +
        ctx.red("RED") +
        ctx.bold("BOLD") +
        "END"
    )
  ).toMatchSnapshot();
});
