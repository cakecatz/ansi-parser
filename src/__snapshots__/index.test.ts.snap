// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bold modifier 1`] = `
Array [
  Object {
    "bg": 0,
    "bold": true,
    "content": "hello, world",
    "dim": false,
    "fg": 0,
    "inverse": false,
    "italic": false,
    "underline": false,
  },
]
`;

exports[`inverse modifier 1`] = `
Array [
  Object {
    "bg": 0,
    "bold": false,
    "content": "hello, world",
    "dim": false,
    "fg": 0,
    "inverse": true,
    "italic": false,
    "underline": false,
  },
]
`;

exports[`italic modifier 1`] = `
Array [
  Object {
    "bg": 0,
    "bold": false,
    "content": "hello, world",
    "dim": false,
    "fg": 0,
    "inverse": false,
    "italic": true,
    "underline": false,
  },
]
`;

exports[`multiple modifiers and colors 1`] = `
Array [
  Object {
    "bg": 0,
    "bold": false,
    "content": "START",
    "dim": false,
    "fg": 0,
    "inverse": false,
    "italic": false,
    "underline": false,
  },
  Object {
    "bg": 0,
    "bold": false,
    "content": "INVERSE",
    "dim": false,
    "fg": 0,
    "inverse": true,
    "italic": false,
    "underline": false,
  },
  Object {
    "bg": 0,
    "bold": false,
    "content": "BLUE",
    "dim": false,
    "fg": 4,
    "inverse": false,
    "italic": false,
    "underline": false,
  },
  Object {
    "bg": 0,
    "bold": false,
    "content": "UNDERLINE",
    "dim": false,
    "fg": 7,
    "inverse": false,
    "italic": false,
    "underline": true,
  },
  Object {
    "bg": 0,
    "bold": false,
    "content": "RED",
    "dim": false,
    "fg": 1,
    "inverse": false,
    "italic": false,
    "underline": true,
  },
  Object {
    "bg": 0,
    "bold": true,
    "content": "BOLD",
    "dim": false,
    "fg": 7,
    "inverse": false,
    "italic": false,
    "underline": true,
  },
  Object {
    "bg": 7,
    "bold": false,
    "content": "END",
    "dim": false,
    "fg": 0,
    "inverse": false,
    "italic": false,
    "underline": true,
  },
]
`;

exports[`parse string not included ansi codes 1`] = `
Array [
  Object {
    "bg": 0,
    "bold": false,
    "content": "hello,world",
    "dim": false,
    "fg": 0,
    "inverse": false,
    "italic": false,
    "underline": false,
  },
]
`;

exports[`string with ansi color 1`] = `
Array [
  Object {
    "bg": 0,
    "bold": false,
    "content": "hello, world",
    "dim": false,
    "fg": 4,
    "inverse": false,
    "italic": false,
    "underline": false,
  },
]
`;

exports[`underline modifier 1`] = `
Array [
  Object {
    "bg": 0,
    "bold": false,
    "content": "hello, world",
    "dim": false,
    "fg": 0,
    "inverse": false,
    "italic": false,
    "underline": true,
  },
]
`;
